ts/sdk/src/vaultSubscriber.ts

```diff
@@ -0,0 +1,155 @@
+import {
+	BulkAccountLoader,
+	DataAndSlot,
+	NotSubscribedError,
+	PublicKey,
+} from '@drift-labs/sdk';
+import { getVaultAddressSync } from './addresses';
+import { Program } from '@coral-xyz/anchor';
+import StrictEventEmitter from 'strict-event-emitter-types';
+import { EventEmitter } from 'events';
+import { Vault, VaultAccountEvents } from './types/types';
+import { DriftVaults } from './types/drift_vaults';
+import { encodeName } from './name';
+
+export class VaultSubscriber {
+	private program: Program<DriftVaults>;
+	private _isSubscribed: boolean;
+	private pubkey: PublicKey;
+	private account?: DataAndSlot<Vault>;
+
+	private _eventEmitter: StrictEventEmitter<EventEmitter, VaultAccountEvents>;
+	private accountLoader: BulkAccountLoader;
+	private callbackId: string | null = null;
+	private errorCallbackId: string | null = null;
+
+	constructor(
+		program: Program<DriftVaults>,
+		vaultPubKey: PublicKey,
+		accountLoader: BulkAccountLoader
```

Comment:
might be good to parameterize this so that you can use websockets or polling in the future (for now we can just implement polling). that way if we add ws in future, it won't be breaking change